zeus/carbon/__init__.py

```diff
@@ -0,0 +1,21 @@
+"""Carbon intensity providers used for carbon-aware optimizers."""
+
+from zeus.carbon.electricity_maps_carbon_intensity_provider import (
+    ElectricityMapsCarbonIntensityProvider,
+)
+
+import requests
+
+
+def get_ip_lat_long() -> tuple[float, float]:
+    """Retrieve the latitude and longitude of the current IP position."""
+    try:
+        ip_url = "http://ipinfo.io/json"
+        resp = requests.get(ip_url)
+        loc = resp.json()["loc"]
+        lat, long = map(float, loc.split(","))
+        print(f"Retrieve latitude and longitude: {lat}, {long}")
+        return lat, long
+    except Exception as e:
```

Comment:
Catching `Exception` is bad practice: https://pylint.readthedocs.io/en/latest/user_guide/messages/warning/broad-exception-caught.html

Reply:
Specifically for these exceptions related to the network, is the correct way to handle this is to just let the exception bubble up?

I'm thinking that we should leave the exception untouched since it is not our responsibility to ensure that, for example, the user is connected to the internet.

Reply:
Nevermind, just read the comment below. Seems like logger.exception is what I wanted to do.